### DISCUSSION TITLE
How to Autohide the scroll bar of the list box in WPF?

### DISCUSSION THREAD

**Discussion by VahidEra** [dotnet/wpf#7803] [Q&A] [answered]
Hi guys,
I have a listbox in my wpf project and I want to autohide its scrollbar, can I do this?

Like this:

![AnimationF](https://user-images.githubusercontent.com/27216710/236609078-be2a1a77-957d-4a30-82b8-7f048178047d.gif)

**Comment by miloush** [comment, marked as answer]
First of all, there is an accessibility settings in Windows to always show scrollbars and you should respect that if you are implementing this yourself.

An easy way is to set `ScrollViewer.VerticalScrollBarVisibility` to `Auto` on `MouseEnter` and to `Hidden` on `MouseLeave`. However, in the default template, this will change the width of the content as the scrollbar appears and disappears. You might need to change the template of the scrollviewer for the scrollbar to be overlaid on top of the content rather than to be to its side.

However, doesn't the material design already support this in their listboxes?

**Reply by VahidEra** [reply]
@miloush Dear Jan, First, thank you very much for your quick reply. I must say that unfortunately there is no option for autohide in Material Design and this option is only available on scrollviewer.  But in the handycontrol component, this option is active by default on all scrollBar.
Again, thank you very much for your help.

**Reply by miloush** [reply]
ListBox contains a ScrollViewer. So in fact you can use the ScrollViewer that comes in these themes. Both Material Design and Handy Control seem to have the autohiding feature.

Handy control (already styles the scrollbar the way you wanted, you don't need a separate style for it):

```xaml
<ListBox.Resources>
    <Style TargetType="ScrollViewer" BasedOn="{StaticResource ScrollViewerNativeBaseStyle}">
        <Setter Property="hc:ScrollViewerAttach.AutoHide" Value="True" />
    </Style>
</ListBox.Resources>
```

Material Design:

```xaml
<ListBox.Resources>
    <Style TargetType="ScrollViewer" BasedOn="{StaticResource MaterialDesignScrollViewer}">
        <Setter Property="md:ScrollViewerAssist.IsAutoHideEnabled" Value="True" />
    </Style>
</ListBox.Resources>
```

I would like to caution you about mixing several theme libraries together, they might conflict with each other and give you some tough times.

**Reply by VahidEra** [reply]
@miloush woooooooooooooow Dear Jan, this is insane . This is exactly what I was looking for Thank you very much for solving the problem and for your help. As I always said you are the best and the best never rest.
